src/tm-threads.c

```diff
@@ -123,22 +140,11 @@ TmEcode TmThreadsSlotVarRun(ThreadVars *tv, Packet *p, TmSlot *slot)
             TmThreadsSlotProcessPktFail(tv, s, NULL);
             return TM_ECODE_FAILED;
         }
-
-        /* handle new packets */
-        while (tv->decode_pq.top != NULL) {
-            Packet *extra_p = PacketDequeueNoLock(&tv->decode_pq);
-            if (unlikely(extra_p == NULL))
-                continue;
-
-            /* see if we need to process the packet */
-            if (s->slot_next != NULL) {
-                r = TmThreadsSlotVarRun(tv, extra_p, s->slot_next);
```

Comment:
Was this bit connected to the behavior described in the commit for bug 6402? (Just curious)

Reply:
it's a cherry-pick to make the backport a bit easier/cleaner

Reply:
Ah, from the other description, I got curious if this part was also related to that. Thanks :)